tests/integration/nested_polymorphic_test.js

```diff
@@ -0,0 +1,177 @@
+      assert.equal(animals.get('length'), 1);
+
+      let first = animals.objectAt(0);
+      assert.ok(first instanceof Animal, 'first animal is Animal');
+      assert.equal(first.get('name'), 'Mittens', 'lion has correct name');
+      assert.ok(first instanceof Lion, 'first animal is Lion');
+      let lionPreys = first.get('preys');
+      assert.equal(lionPreys.get('length'), 2, 'lion has 2 preys');
+
+      let firstPrey = lionPreys.objectAt(0);
+      assert.ok(firstPrey instanceof PreyGiraffe, 'first prey is Giraffe');
+      assert.equal(firstPrey.get('neckLength'), '20', 'giraffe\'s neck length is correct');
+
+      let secondPrey = lionPreys.objectAt(1);
+      assert.ok(secondPrey instanceof PreyHog, 'second prey is Hog');
+      assert.equal(secondPrey.get('stripesColour'), 'golden', 'hog\'s stripes colour is correct');
+    });
+  });
+
+  test('when only inner level types are normalized', function(assert) {
+    let data = {
+      id: 1,
+      name: 'Nested Zoo',
+      animals: [
+        {
+          $type: 'lion',
+          name: 'Mittens',
+          hasManes: 'true',
+          preys: [
+            {
+              _normalizeBackendTypeTo$Type: true,
+              backendType: 'giraffe',
+              neckLength: 20
+            },
+            {
+              _normalizeBackendTypeTo$Type: true,
+              backendType: 'hog',
+              stripesColour: 'golden'
+            }
+          ]
+        }
+      ]
+    };
+
+    server.get('/zoos/1', () => {
+      return [200, { 'Content-Type': 'application/json' }, JSON.stringify({ zoo: data })];
+    });
+
+    return store.find('zoo', 1).then(zoo => {
+      let animals = zoo.get('animals');
+      assert.equal(animals.get('length'), 1);
+
+      let first = animals.objectAt(0);
+      assert.ok(first instanceof Animal, 'first animal is Animal');
+      assert.equal(first.get('name'), 'Mittens', 'lion has correct name');
+      assert.ok(first instanceof Lion, 'first animal is Lion');
+      let lionPreys = first.get('preys');
+      assert.equal(lionPreys.get('length'), 2, 'lion has 2 preys');
+
+      let firstPrey = lionPreys.objectAt(0);
+      assert.ok(firstPrey instanceof PreyGiraffe, 'first prey is Giraffe');
+      assert.equal(firstPrey.get('neckLength'), '20', 'giraffe\'s neck length is correct');
+
+      let secondPrey = lionPreys.objectAt(1);
+      assert.ok(secondPrey instanceof PreyHog, 'second prey is Hog');
+      assert.equal(secondPrey.get('stripesColour'), 'golden', 'hog\'s stripes colour is correct');
+    });
+  });
+
+  test('when only outer level types are normalized', function(assert) {
+    let data = {
+      id: 1,
+      name: 'Nested Zoo',
+      animals: [
+        {
+          _normalizeBackendTypeTo$Type: true,
+          backendType: 'lion',
+          name: 'Mittens',
+          hasManes: 'true',
+          preys: [
+            {
+              $type: 'prey-giraffe',
+              neckLength: 20
```

Comment:
This case works correctly, as the normalization is not needed to be nested. It seems like the normalization is only respected on the root model level (zoo), and not correctly recursively applied to the nested fragments.